### DISCUSSION TITLE
How to create a diagram with this type of process in Joint js

### DISCUSSION THREAD

**Discussion by nikita-fxbytes** [clientIO/joint#2513] [Q&A]
### Introduction

**i want it like this**
1. The header should contain buttons for all shapes. On clicking the shape that shape will appear on the paper with ui.Halo

2. Whenever I click on a shape, that shape appears in the side bar with its own text.

3. Whichever shape I click on, its ui.Inspector appears.

4. If I drag and drop a shape on the paper, the position of the shape in the side bar will change and if any shape has a relationship, it will also appear on the side as if it were a child or a parent.

5. I can edit, delete shapes

How do I put it all together?

i want this type

![image_2024_01_30T05_40_14_455Z](https://github.com/clientIO/joint/assets/99159940/e0a66f75-83ea-4df3-b944-e4f6d36b3ddd)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by nikita-fxbytes** [comment]
Hello
Please help me, I urgently need assistance.

**Comment by kumilingus** [comment]
> The header should contain buttons for all shapes. On clicking the shape that shape will appear on the paper with ui.Halo

See our *Stencil* [documentation](https://resources.jointjs.com/docs/rappid/v4.0/ui.Stencil.html#ui.Stencil) and this [diccussion](https://github.com/clientIO/joint/discussions/2472),

> Whenever I click on a shape, that shape appears in the side bar with its own text.

Perhaps this [example](https://www.jointjs.com/demos/diagram-index) will help.

> Whichever shape I click on, its ui.Inspector appears.

To find more info on *Inspector* please see our [documentation](https://resources.jointjs.com/docs/rappid/v4.0/ui.Inspector.html) or this simple [example](https://codepen.io/jointjs/pen/qBoqaep/3ebc14cfce30e8a454e0b7b3f06a1b2e).

> If I drag and drop a shape on the paper, the position of the shape in the side bar will change and if any shape has a relationship, it will also appear on the side as if it were a child or a parent.

You can listen to various [events](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#mvc.Events.events) on graph and re-render your sidebar.

`graph.getElements()` gives you an array of all elements in the graph.
`graph.getNeighbors(el1, { outbound: true })` gives you an array of elements connected to `el1` where `el1` is the source.

> I can edit, delete shapes

You can edit your shapes via *Inspector*.
You can delete any shape with [element.remove()](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Element.prototype.remove).

If you want to delete shapes using keyboard, see our [Keyboard](https://resources.jointjs.com/docs/rappid/v4.0/ui.Keyboard.html) plugin.

> Please help me, I urgently need assistance.

Please see our [paid support package]( https://www.jointjs.com/support) for guaranteed response time and more.
